src/main/java/io/appium/java_client/pagefactory/AppiumElementLocator.java

```diff
@@ -46,10 +48,36 @@ class AppiumElementLocator implements CacheableLocator {
     private final boolean shouldCache;
     private final By by;
     private final Duration duration;
+    private final WeakReference<SearchContext> searchContextReference;
     private final SearchContext searchContext;
+
     private WebElement cachedElement;
     private List<WebElement> cachedElementList;
 
+    /**
+     * Creates a new mobile element locator. It instantiates {@link WebElement}
+     * using @AndroidFindBy (-s), @iOSFindBy (-s) and @FindBy (-s) annotation
+     * sets
+     *
+     * @param searchContextReference     The context reference to use when finding the element
+     * @param by                         a By locator strategy
+     * @param shouldCache                is the flag that signalizes that elements which
+     *                                   are found once should be cached
+     * @param duration                   timeout parameter for the element to be found
+     */
+    public AppiumElementLocator(
```

Comment:
```suggestion
    AppiumElementLocator(
```
does it make sense to make this constructor package-private?

Reply:
The same quesiton for all new API-s using `WeakReference`-s

Reply:
changed